moveit_core/collision_detection_fcl/src/collision_common.cpp

```diff
@@ -550,62 +534,63 @@ bool distanceCallback(fcl::CollisionObject* o1, fcl::CollisionObject* o2, void*
         int max_index = 0;
         for (int i = 0; i < contacts; ++i)
         {
-          const fcl::Contact& contact = coll_res.getContact(i);
+          const fcl::Contact &contact = coll_res.getContact(i);
           if (contact.penetration_depth > max_dist)
           {
             max_dist = contact.penetration_depth;
             max_index = i;
           }
         }
 
-        const fcl::Contact& contact = coll_res.getContact(max_index);
+        const fcl::Contact &contact = coll_res.getContact(max_index);
         dist_result.distance = -contact.penetration_depth;
         dist_result.nearest_points[0] = Eigen::Vector3d(contact.pos.data.vs);
         dist_result.nearest_points[1] = Eigen::Vector3d(contact.pos.data.vs);
         dist_result.normal = Eigen::Vector3d(contact.normal.data.vs);
       }
     }
 
-    cdata->res->minimum_distance = dist_result;
+    if (dist_result.distance < cdata->res->minimum_distance.distance)
+    {
+      cdata->res->minimum_distance = dist_result;
+    }
+
+    if (dist_result.distance <= 0 && !cdata->res->collision)
+    {
+      cdata->res->collision = true;
+    }
 
-    if (!cdata->req->global_minimum_only)
+    if (cdata->req->type != DistanceRequestType::GLOBAL)
     {
-      if (d <= 0 && !cdata->res->collision)
+      if (it == cdata->res->distances.end())
       {
-        cdata->res->collision = true;
+        std::vector<DistanceResultsData> data;
+        data.reserve(cdata->req->max_contacts_per_body);
```

Comment:
i don't understand why you reserve this vector then only ``push_back`` one item. i don't see a for loop anywhere else to fill this in.

Reply:
The loop is hidden within fcl, it calls the callback function multiple times.

Reply:
please add this as a comment since its not clear why ``reserve`` is otherwise needed and it would run fine without it (its a hidden optimization)